code/game/objects/items/ego_weapons/ordeal.dm

```diff
@@ -0,0 +1,790 @@
+							TEMPERANCE_ATTRIBUTE = 100,
+							JUSTICE_ATTRIBUTE = 100
+							)
+	var/mob/current_holder
+	var/form = "red"
+	var/index = 1
+	var/special_attack = FALSE
+	var/list/weapon_list = list(
+		"red" = list(70, RED_DAMAGE, "This weapon changes its damage type every 2 minutes. Use in hand to open a portal in front of you releasing a RED damage claw.", 240),
+		"white" = list(70, WHITE_DAMAGE, "This weapon changes its damage type every 2 minutes. Use in hand to open a portal at your location releasing a WHITE damage tentacle.", 240),
+		"black" = list(60, BLACK_DAMAGE, "This weapon changes its damage type every 2 minutes. Use in hand to open multiple portals around you which unleashes a BLACK spike in a 5x5 aoe.", 180),
+		"pale" = list(50, PALE_DAMAGE, "This weapon changes its damage type every 2 minutes. Use in hand to summon a stationary pulsing PALE damage eye.", 60)
+		)
+	var/special_cooldown
+	var/special_cooldown_time = 30 SECONDS
+	var/special_checks_faction = TRUE
+	var/special_damage = 240
+	var/switch_cooldown_time = 2 MINUTES // Workshop this value.
+	var/list/intrusive_thoughts = list(
+		"They could never be understood, we ignored their warnings and warped our minds.",
+		"The idea of Love and Compassion was alien to them, all they could give was the cold reality.",
+		"We built shrines and scribed their languages in their image, they ignored our pleas and left nothing.",
+		"We damned the consequences, and spiraled into insanity."
+	)
+
+/obj/item/ego_weapon/god_delusion/Initialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+	addtimer(CALLBACK(src, .proc/Transform), switch_cooldown_time) // Kicks off the changing
+
+/obj/item/ego_weapon/god_delusion/equipped(mob/user, slot)
+	. = ..()
+	if(!user)
+		return
+	current_holder = user
+
+/obj/item/ego_weapon/god_delusion/dropped(mob/user)
+	. = ..()
+	current_holder = null
+
+/obj/item/ego_weapon/god_delusion/attack_self(mob/user)
+	. = ..()
+	if(!CanUseEgo(user))
+		return
+	if(special_cooldown > world.time)
+		to_chat(user, "<span class='warning'>Your ability is on cooldown!</span>")
+		return
+	special_cooldown = world.time + special_cooldown_time
+	switch(form)
+		if("red")
+			special_attack = !special_attack
+			if(special_attack)
+				to_chat(user, "<span class='notice'>You prepare your portal.</span>")
+			else
+				to_chat(user, "<span class='notice'>You decide to not use the red claw.</span>")
+		if("white")
+			to_chat(user, "<span class='warning'>Not done</span>")
+		if("black")
+			for(var/turf/L in view(2, user))
+				new /obj/effect/temp_visual/cult/sparks(L)
+			playsound(user, 'sound/effects/ordeals/violet/midnight_portal_on.ogg', 50, FALSE)
+			if(!do_after(user, 1 SECONDS, src))
+				return
+			playsound(user, 'sound/effects/ordeals/violet/midnight_black_attack1.ogg', 50, FALSE)
+			for(var/turf/open/T in view(2, user))
+				new /obj/effect/temp_visual/small_smoke/halfsecond(T)
+			for(var/mob/living/L in livinginrange(2, user))
```

Comment:
`livinginrange` is for large areas when you want to make sure people don't hide in lockers.
Just use normal range here.

Reply:
Hell, use view, just like you did for the visual effect.